File: BowlingKata_Java/app/src/main/java/bowling_kata/Game.java
package bowling_kata;

public class Game {

    private int rolls[] = new int[21];
    private int currentRoll = 0;
    private int score = 0;

    private boolean isStrike(int rollIndex) {
        return rolls[rollIndex] == 10;
    }

    private boolean isSpare(int rollIndex) {
        return rolls[rollIndex] + rolls[rollIndex + 1] == 10;
    }

    private int strikeBonus(int rollIndex) {
        return rolls[rollIndex + 1] + rolls[rollIndex + 2];
    }

    private int spareBonus(int rollIndex) {
        return rolls[rollIndex + 2];
    }

    private int normalFrameScore(int rollIndex) {
        return rolls[rollIndex] + rolls[rollIndex + 1];
    }

    public void roll(int pins) {
        rolls[currentRoll++] = pins;
    }

    public int score() {
        int rollIndex = 0;

        for (int frame = 0; frame < 10; frame++) { // Iterate over frames
            if (isStrike(rollIndex)) {
                score += 10 + strikeBonus(rollIndex);
                rollIndex++;
            }
            else if (isSpare(rollIndex)) {
                score += 10 + spareBonus(rollIndex);
                rollIndex += 2;
            }
            else {
                score += normalFrameScore(rollIndex);
                rollIndex += 2;
            }
        }
        return score;
    }

    public static void main(String[] args) {
        System.out.println("Game class executed!");
    }
}

File: BowlingKata_TS/src/Game.ts
export class Game {
    private rolls: number[] = [21];
    private currentRoll: number = 0;
    private gameScore: number = 0;

    private isStrike(rollIndex: number): boolean {
        return this.rolls[rollIndex] == 10;
    }

    private isSpare(rollIndex: number): boolean {
        return this.rolls[rollIndex] + this.rolls[rollIndex + 1] == 10;
    }

    private strikeBonus(rollIndex: number): number {
        return this.rolls[rollIndex + 1] + this.rolls[rollIndex + 2];
    }

    private spareBonus(rollIndex: number): number {
        return this.rolls[rollIndex + 2];
    }

    private normalFrameScore(rollIndex: number): number {
        return this.rolls[rollIndex] + this.rolls[rollIndex + 1];
    }

    public roll(pins: number) {
        this.rolls[this.currentRoll++] = pins;
    }

    public score(): number {
        let rollIndex = 0;

        for (let frame = 0; frame < 10; frame++) { // Iterate over frames
            if (this.isStrike(rollIndex)) {
                this.gameScore += 10 + this.strikeBonus(rollIndex);
                rollIndex++;
            } else if (this.isSpare(rollIndex)) {
                this.gameScore += 10 + this.spareBonus(rollIndex);
                rollIndex += 2;
            } else {
                this.gameScore += this.normalFrameScore(rollIndex);
                rollIndex += 2;
            }
        }
        return this.gameScore;
    }
}
